specutils/io/default_loaders/hst_stis.py

```diff
@@ -1,57 +1,175 @@
-        dispersion = hdulist[1].data['wavelength'].flatten() * disp_unit
-        uncertainty = StdDevUncertainty(hdulist[1].data["ERROR"].flatten() * unit)
-
-    sort_idx = dispersion.argsort()
-    dispersion = dispersion[sort_idx]
-    data = data[sort_idx]
-    uncertainty = uncertainty[sort_idx]
-
-    return Spectrum1D(flux=data,
-                      spectral_axis=dispersion,
-                      uncertainty=uncertainty,
-                      meta=meta)
+        if len(hdulist) <= 1:
+            raise ValueError('HST/STIS file is mising the SCI data extension.')
+        if len(hdulist) > 2:
+            raise RuntimeError('HST/STIS file is multi-extension.  '
+                               'Use SpectrumCollection.read() instead.')
+        if len(hdulist[1].data) > 1:
+            raise RuntimeError('HST/STIS file has multiple orders (echelle data).  '
+                               'Use SpectrumCollection.read() instead.')
+
+        # Extract the single spectrum returned from the list (a Spectrum1D object):
+        return _construct_Spectrum1D_list(hdulist, sdqflags=sdqflags)[0]
+
+
+@data_loader(
+    label="HST/STIS multi", identifier=identify_stis_multiorder_or_multiread,
+    extensions=['FITS', 'FIT', 'fits', 'fit'], priority=10,
+    dtype=SpectrumCollection,
+)
+def stis_multi_spectrum_loader(file_obj, **kwargs):
+    """Load STIS spectral data from the MAST archive into a spectrum object.
+
+    Parameters
+    ----------
+    file_obj: str, file-like, or HDUList
+          FITS file name, object (provided from name by Astropy I/O Registry),
+          or HDUList (as resulting from astropy.io.fits.open()).
+
+    sdqflags: int or None
+          Serious data quality flag bit-wise mask to apply.
+          If None, use the default value found in the SCI ext headers.
+
+    Returns
+    -------
+    data: SpectrumCollection
+        The spectra that are represented by the data in this table.
+
+    Note
+    ----
+    See here for the definitions of the STIS data quality (DQ) bits:
+    https://hst-docs.stsci.edu/stisdhb/chapter-2-stis-data-structure/2-5-error-and-data-quality-array#id-2.5ErrorandDataQualityArray-2.5.2DataQualityFlagging
+    """
+    sdqflags = kwargs.pop('sdqflags', None)
+
+    with read_fileobj_or_hdulist(file_obj, **kwargs) as hdulist:
+        if len(hdulist) <= 1:
+            raise ValueError('HST/STIS file is mising the SCI data extension.')
+        if (len(hdulist) == 2) and (len(hdulist[1].data) == 1):
+            raise RuntimeError('HST/STIS file is single-read and first-order.  '
+                               'Use Spectrum1D.read() instead.')
+
+        multi_spec = _construct_Spectrum1D_list(hdulist, sdqflags=sdqflags)
+
+    return SpectrumCollection.from_spectra(multi_spec)
+
+
+def _construct_Spectrum1D_list(hdulist, sdqflags=None):
+    """Construct a list of Spectrum1D objects from an HST/STIS FITS HDUList.
+    """
+    meta = {'header_primary': hdulist[0].header}
```

Comment:
I just noticed that the `IRAF` FITS write assumes the primary header is included as "header".  Should I change this?

Reply:
Will do.

There can be some discrepancy between the primary and SCI headers for multi-read STIS observation.  E.g., the start time listed in the primary header won't match the second+ SCI read, primary header's exposure time is the sum of the SCI exposures.  I think we'd need complex logic to combine them, so I'm happy keeping them separate.